.eslintrc.js

```diff
@@ -84,106 +115,90 @@ module.exports = {
         {
             files: ["lib/*"],
             rules: {
-                "no-restricted-modules": ["error", {
-                    patterns: [
-                        ...internalFiles
-                    ]
-                }]
+                "node/no-restricted-require": ["error", [
+                    ...createInternalFilesPatterns()
+                ]]
             }
         },
         {
-            files: ["lib/cli-engine/**/*"],
+            files: [INTERNAL_FILES.CLI_ENGINE_PATTERN],
             rules: {
-                "no-restricted-modules": ["error", {
-                    patterns: [
-                        ...internalFiles,
-                        "**/init"
```

Comment:
I'm not sure why we specifically disallow these directory imports (`init/index.js`, etc.). Do we want to continue doing this? It feels a bit overkill to me, but maybe I'm missing some context.

Reply:
That's to prevent depending `fs` accidentally and also prevent cyclic depending. The diagram in the [Architecture](https://eslint.org/docs/developer-guide/architecture) page describes the depending directions of each directory.

Reply:
Thanks for the explanation.